registry/retriever/extrinsic_retriever_live_test.go

```diff
@@ -0,0 +1,198 @@
+	"github.com/centrifuge/go-substrate-rpc-client/v4/rpc/chain/generic"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/scale"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
+)
+
+var (
+	extTestChains = []testFnProvider{
+		&testChain[
+			types.MultiAddress,
+			types.MultiSignature,
+			generic.DefaultPaymentFields,
+		]{
+			url: "wss://fullnode.parachain.centrifuge.io",
+		},
+		&testChain[
+			types.MultiAddress,
+			types.MultiSignature,
+			generic.DefaultPaymentFields,
+		]{
+			url: "wss://rpc.polkadot.io",
+		},
+		&testChain[
+			types.MultiAddress,
+			types.MultiSignature,
+			generic.PaymentFieldsWithAssetID,
+		]{
+			url: "wss://statemint-rpc.polkadot.io",
+		},
+		&testChain[
+			types.MultiAddress,
+			AcalaMultiSignature,
+			generic.DefaultPaymentFields,
+		]{
+			url: "wss://acala-rpc-0.aca-api.network",
+		},
+		&testChain[
+			[20]byte,
+			[65]byte,
+			generic.DefaultPaymentFields,
+		]{
+			url: "wss://wss.api.moonbeam.network",
+		},
+	}
+)
+
+func TestLive_ExtrinsicRetriever_GetExtrinsics(t *testing.T) {
+	t.Parallel()
+
+	var wg sync.WaitGroup
+
+	for _, c := range extTestChains {
+		c := c
+
+		wg.Add(1)
+
+		go c.GetTestFn()(&wg)
+	}
+
+	wg.Wait()
+}
+
+type testFn func(wg *sync.WaitGroup)
+
+type testFnProvider interface {
+	GetTestFn() testFn
+}
+
+type testChain[
+	A any,
+	S any,
+	P any,
+] struct {
+	url string
+}
+
+func (t *testChain[A, S, P]) GetTestFn() testFn {
+	return func(wg *sync.WaitGroup) {
+		defer wg.Done()
+
+		testURL := t.url
+
+		api, err := gsrpc.NewSubstrateAPI(testURL)
+
+		if err != nil {
+			log.Printf("Couldn't connect to '%s': %s\n", testURL, err)
+			return
+		}
+
+		chain := generic.NewChain[A, S, P, *generic.SignedBlock[A, S, P]](api.Client)
+
+		extrinsicRetriever, err := NewDefaultExtrinsicRetriever[A, S, P](chain, api.RPC.State)
+
+		if err != nil {
+			log.Printf("Couldn't create extrinsic retriever: %s", err)
+			return
+		}
+
+		header, err := api.RPC.Chain.GetHeaderLatest()
+
+		if err != nil {
+			log.Printf("Couldn't get latest header for '%s': %s\n", testURL, err)
+			return
+		}
+
+		processedBlockCount := 0
+
+		for {
+			blockHash, err := api.RPC.Chain.GetBlockHash(uint64(header.Number))
+
+			if err != nil {
+				log.Printf("Couldn't retrieve blockHash for '%s', block number %d: %s\n", testURL, header.Number, err)
+				return
+			}
+
+			_, err = extrinsicRetriever.GetExtrinsics(blockHash)
+
+			if err != nil {
+				log.Printf("Couldn't retrieve extrinsics for '%s', block number %d: %s\n", testURL, header.Number, err)
+			}
+
+			header, err = api.RPC.Chain.GetHeader(header.ParentHash)
+
+			if err != nil {
+				log.Printf("Couldn't retrieve header for block number '%d' for '%s': %s\n", header.Number, testURL, err)
+
+				return
+			}
+
+			processedBlockCount++
+
+			if processedBlockCount%500 == 0 {
+				log.Printf("Retrieved calls for %d blocks for '%s' so far, last block number %d\n", processedBlockCount, testURL, header.Number)
+			}
+		}
+	}
```

Comment:
Initially this was just a test that would attempt to go to 0 to be ultra-sure that decoding works properly. Even though in theory that wouldn't work since we would reach metadata v13 which would cause the whole thing to fail.

I like the idea of having a max number of extrinsics/events parsed or similar.